firmware/broadcom_trx.ksy

```diff
@@ -134,9 +159,44 @@ types:
             pos: ofs_body
             size: len_body
             if: is_present
+          usual_purpose:
+            value: idx
+            enum: usual_purpose
+        enums:
+          usual_purpose:
+            0xFF: unknown
+            0: bootloader
+            1: kernel
+            2: rootfs
+            3: some_header
+
       flags:
+        doc: the the names of these flags will probably be changed
         seq:
-          - id: flags
+          - id: unkn
```

Comment:
>just omit the id instead of naming it unknown

It harms the goal of eventually reversing it. That's why, even if KS allows omitting `id`, I always assign it, and also in the changes I have proposed into the style guide I added some words about how to name them.